hyperspy/axes.py

```diff
@@ -400,6 +367,240 @@ def _get_index(self, value):
+                 units=None,
+                 navigate=False,
+                 is_binned=False,
+                 axis=[1],
+                 **kwargs):
+        super().__init__(
+            index_in_array=index_in_array,
+            name=name,
+            units=units,
+            navigate=navigate,
+            is_binned=is_binned,
+            **kwargs)
+        self.axis = axis
+
+    def _slice_me(self, slice_):
+        """Returns a slice to slice the corresponding data axis and set the
+        axis accordingly.
+
+        Parameters
+        ----------
+        slice_ : {int, slice}
+
+        Returns
+        -------
+        my_slice : slice
+
+        """
+        my_slice = self._get_array_slices(slice_)
+        self.axis = self.axis[my_slice]
+        return my_slice
+
+
+    def get_axis_dictionary(self):
+        d = super().get_axis_dictionary()
+        d.update({'axis': self.axis})
+        return d
+
+    @property
+    def low_value(self):
+        if self._is_increasing_order is True:
+            return self.axis[0]
+        elif self._is_increasing_order is False:
+            return self.axis[-1]
+        else:
+            raise NotImplementedError("The axis is not ordered so there is no high/low value")
+    @property
+    def value(self):
+        return self.axis[self.index]
+
+    @value.setter
+    def value(self, val):
+        self.index = self.value2index(val)
+
+    @property
+    def high_value(self):
+        if self._is_increasing_order is True:
+            return self.axis[-1]
+        elif self._is_increasing_order is False:
+            return self.axis[0]
+        else:
+            raise NotImplementedError("The axis is not ordered so there is no high/low value")
+
+    @property
+    def index(self):
+        return self._index
+
+    @index.setter
+    def index(self, value):
+        if self._index != value:
+            if self.low_index <= value <= self.high_index:
+                self._index = value
+                self.events.index_changed.trigger(obj=self, index=self.index)
+
+    @property
+    def low_index(self):
+        return 0
+
+    @property
+    def high_index(self):
+        return self.size - 1
+
+    @property
+    def size(self):
+        return len(self.axis)
+
+    @property
+    def _is_increasing_order(self):
+        """
+        Determine if the axis has an increasing, decreasing order or no order
+        at all.
+
+        Returns
+        -------
+        True if order is increasing, False if order is decreasing, None
+        otherwise.
+
+        """
+        try:
+            steps = self.axis[1:] - self.axis[:-1]
+            if np.all(steps > 0):
+                return True
+            elif np.all(steps < 0):
+                return False
+            else:
+                # the axis is not ordered
+                return None
+        except np.core._exceptions._UFuncNoLoopError:  # subtracting array of objects/strings will fail
```

Comment:
Avoid using numpy private API: `np.core._exceptions._UFuncNoLoopError` is a subclass of `TypeError`:
```suggestion
        except TypeError:  # subtracting array of objects/strings will fail
```

Reply:
@ericpre Good point! I'll make the substitution